### ISSUE TITLE
Invocation.thenReturn() may fail with varargs (similar to issue 123)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dlsgusrn7577/mockito#157]
```
What steps will reproduce the problem?
1. Has a class with a method like this: blah(Object... o)
2. Stub it like this: when(mock.blah(anyVararg())).thenReturn(-1)
3. Run test.

What is the expected output? What do you see instead?
It should return -1 on both calls, instead it threw an 
ArrayIndexOutOfBoundsException on the second assert

What version of the product are you using? On what operating system?
1.8.1
MacOS 10.5.8

Please provide any additional information below.
    class Foo {
        int blah(Object... o) {
            return 0;
        }
    }

    @Test
    public void shouldNotThrowAnyException() throws Exception {
        Foo foo = mock(Foo.class);

        when(foo.blah(anyVararg())).thenReturn(-1);

        assertEquals(-1, foo.blah("foo"));
        assertEquals(-1, foo.blah());
    }

```

Original issue reported on code.google.com by `Greg.Hen...@gmail.com` on 10 Dec 2009 at 11:02

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r1790.
```

Original comment by `szcze...@gmail.com` on 11 Dec 2009 at 8:30
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Ouch... Thanks a lot for reporting! 

It's fixed in trunk, will try to get it out soon.
```

Original comment by `szcze...@gmail.com` on 11 Dec 2009 at 8:32

**Comment by GoogleCodeExporter** [comment]
```
looks great - looking forward to the release.  thanks for the quick fix!
```

Original comment by `Greg.Hen...@gmail.com` on 11 Dec 2009 at 8:49

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 12 Dec 2009 at 2:20
- Added labels: **Milestone-Release1.8.2**

**Comment by GoogleCodeExporter** [comment]
```
Released in 1.8.2 on 12-12-2009
```

Original comment by `szcze...@gmail.com` on 13 Dec 2009 at 10:41
